pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -237,15 +252,22 @@ func getIDsFromContainerReferences(cluster cluster.Cluster, tenantId string, con
 	}
 	// loop through all the containers to find the containerIds that at associated with the container references.
 	// eligible containers must belong to the tenant
+	foundCntDown := len(containerReferences)
+Loop:
 	for _, container := range cluster.Containers() {
+		log.Debugf("getIDsFromContainerReferences check container: %+v", container)
+
 		if container.Labels[headers.TenancyLabel] == tenantId {
 			var err error
 			var containerId string
 			// look for containerReference found in volumes_from and links
 			for _, containerReference := range containerReferences {
 				if containerId, err = getID(container, tenantId, containerReference); err == nil {
 					containerReferenceToIdMap[containerReference] = containerId
-					break
+					foundCntDown--
```

Comment:
Why do you need this condition ?   In any case you loop only till containerReferences ??